testing/dbtest/template.go

```diff
@@ -0,0 +1,111 @@
+package dbtest
+
+import (
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"time"
+
+	// postgres dialect
+	_ "github.com/lib/pq"
+)
+
+var (
+	// StartUsingTemplate creates a new test database from a postgres template database.
+	StartUsingTemplate func(dialect string, opt ...Option) (func() error, string, string, error) = startUsingTemplate
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyz"
+const (
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+func randStr(n int) string {
+	b := make([]byte, n)
```

Comment:
is n == 0 a valid parameter?

Reply:
Hmm...its "valid" in that it works and returns a string of length 0. I don't know why anyone would want to do that though. But given that this is an un-exported function used to just create a random suffix for a test database I don't know if we need a check for that case.

Also, if we already have something to generate a random string that is efficient and doesn't need crypto, I can use that instead of defining this function here, but I wasn't able to find one when I looked.

Reply:
Take a look at ` base62.Random` to see if it fits your requirements.   

When it's cheap (perf wise) to do, then adding a check for invalid params always feels like the right pattern to follow, even for un-exported funcs.  BTW, I agree that passing zero makes no sense... unfortunately, I'm one of those devs that occasionally writes code inadvertently that makes no sense.

Reply:
discussed over slack. ty!